File: leetcode-js/357-count-numbers-with-unique-digits.js
/**
 * @param {number} n
 * @return {number}
 */
var countNumbersWithUniqueDigits = function(n) {
    if (n === 0) {
        return 1;
    }

    let res = 10;
    let uniqueDigits = 9;
    let availableNumber = 9;

    while (n > 1 && availableNumber > 0) {
        uniqueDigits *= availableNumber;
        res += uniqueDigits;
        availableNumber -= 1;
        n -= 1;
    }

    return res;
};

File: leetcode-java/357-count-numbers-with-unique-digits.java
public class Solution {
    public int countNumbersWithUniqueDigits(int n) {
        if (n == 0) {
            return 1;
        }

        int res = 10;
        int remaining = 9;
        int unique = 9;

        while (n > 1 && remaining > 0) {
            unique *= remaining;
            res += unique;
            remaining--;
            n--;
        }

        return res;
    }
}
